packages/genesys-spark-components/src/components/beta/gux-rich-text-editor/gux-rich-text-editor-action/gux-rich-text-editor-action.tsx

```diff
@@ -0,0 +1,67 @@
+import { Component, Element, Prop, h } from '@stencil/core';
+import { trackComponent } from '@utils/tracking/usage';
+
+import { buildI18nForComponent, GetI18nValue } from 'i18n';
+import translationResources from './i18n/en.json';
+import { GuxRichTextEditorActionTypes } from './gux-rich-text-editor-action.types';
+import { calculateDisabledState } from '../gux-rich-text-editor.service';
+import { returnActionTypeIcon } from '../gux-rich-text-editor.service';
+
+@Component({
+  tag: 'gux-rich-text-editor-action',
+  styleUrl: 'gux-rich-text-editor-action.scss',
+  shadow: true
+})
+export class GuxRichTextEditorAction {
+  private i18n: GetI18nValue;
+
+  @Element()
+  root: HTMLElement;
+
+  @Prop()
+  action: GuxRichTextEditorActionTypes;
+
+  @Prop({ mutable: true })
+  disabled: boolean = false;
+
+  @Prop()
+  isActive: boolean = false;
+
+  async componentWillLoad(): Promise<void> {
+    trackComponent(this.root, { variant: this.action });
+    this.i18n = await buildI18nForComponent(this.root, translationResources);
+    this.disabled = calculateDisabledState(this.root);
+  }
+
+  private renderTooltip(): JSX.Element {
+    if (!this.disabled) {
+      return (
+        <gux-tooltip>
+          <div slot="content">{this.i18n(this.action)}</div>
+        </gux-tooltip>
+      ) as JSX.Element;
+    }
+  }
+
+  private renderActionButton(): JSX.Element {
+    return (
+      <gux-button-slot accent="ghost" icon-only>
+        <button
+          type="button"
+          disabled={this.disabled}
+          class={{ 'gux-is-active': this.isActive }}
```

Comment:
You will probably need aria-selected or aria-pressed here to denote the active state.